front/pages/w/[wId]/builder/data-sources/managed.tsx

```diff
@@ -518,8 +521,7 @@ export default function DataSourcesView({
             .map((ds) => {
               const isBuilt =
                 ds.status === "built" ||
-                (ds.status === "built-dust-only" &&
-                  isDevelopmentOrDustWorkspace(owner));
+                (ds.connectorProvider === "intercom" && isIntercomEnabled);
```

Comment:
Could potentially have this be slightly more generic by having the status be "built" but have an optional "feature_flag" field on the CONNECTOR_CONFIGURATIONS.

If the FF field is non-null, then we check if the FF is enabled for the workspace. WDYT ?

Reply:
I added a commit to be more generic, but with default to preview to make sure we never roll out by mistake. WDYT?

Reply:
Sounds good, there's a tiny amount of type-safety we could add to make sure that if `preview`, `rollingOutFlag` has to be non-null (and the other way around), but this is very nit :)